src/app/components/container/container.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Input, OnInit, Output } from '@angular/core';
+import { PokemonService } from '../../services/pokemon.service';
+import { Observable } from 'rxjs';
+import Pokemon from '../../models/Pokemon';
+
+@Component({
+  selector: 'app-container',
+  templateUrl: './container.component.html',
+  styleUrls: ['./container.component.css']
+})
+
+/**
+ * @class ContainerComponent
+ * @description This component is responsible for the initial loding of the pokemon list and for the pagination
+ * @property { boolean } loading - The loading state of the component
+ * @property { boolean } loadingChange - The loading state of the component
+ * @property { Observable<Pokemon[]> } pokemons - The list of pokemons
+ * 
+ * @example <app-container></app-container>
+ * 
+ * @method ngOnInit - This method is responsible for initializing the component
+ * 
+ */
+export class ContainerComponent implements OnInit {
+  @Input() loading: boolean;
+  @Output() loadingChange: boolean;
+  pokemons : Observable<Pokemon[]> = new Observable<Pokemon[]>();
+
+  constructor(private pokemonService : PokemonService) {
+    this.loading = true;
+    this.loadingChange = false;
+  }
+
+  ngOnInit(): void {
+    this.pokemons = this.pokemonService.getPokemons();
+    this.loading = false;
+    this.loadingChange = true;
```

Comment:
potresti usare solo `this.loading` e `!this.loading` al posto di usare due variabili (?)

Reply:
poi gli output di solito sono event emitter non variabili in se, non li ho mai usati come variabili in sè

Reply:
Non ho ancora capito come usarli bene. Credo servano per fare 2 way binding, che penso abbia senso nel container perchè dovrebbe comunicare con il componente dell'app e rispondere di conseguenza ad ogni differente ricerca. La segno comunque per la prossima visto che penso andrà riscritta quando mi servirà